net/quic/socket.c

```diff
@@ -1639,18 +1637,18 @@ static void quic_shutdown(struct sock *sk, int how)
 	inet_sk_set_state(sk, QUIC_SS_CLOSED);
 }
 
-static int quic_ioctl(struct sock *sk, int cmd, int *karg)
+static int quic_ioctl(struct sock *sk, int cmd, unsigned long arg)
 {
```

Comment:
I was using the latest upstream kernel, it has been changed to "int (*ioctl)(struct sock *sk, int cmd, int *karg)" in kernel 6.4 in this commit:

https://lore.kernel.org/lkml/20230609200010.27991-1-kuniyu@amazon.com/T/

This quic implementation is not in the upstream kernel yet and for now we expect this to be used in old kernels, and also it's difficult to detect ioctl format only by `#if KERNEL_VERSION()`  as some release might backport it to their old version kernels.

Can you completely revert/delete this ioctl support feature in QUIC in this patch for now?

Thanks.